crates/tx/src/types.rs

```diff
@@ -909,6 +899,61 @@ pub struct Header {
     /// In case a memo is not present in the transaction, a
     /// byte array filled with zeroes is present instead
     pub memo_hash: namada_core::hash::Hash,
+}
+
+impl TxCommitments {
+    /// Get the hash of this transaction's code
+    pub fn code_sechash(&self) -> &namada_core::hash::Hash {
+        &self.code_hash
+    }
+
+    /// Get the transaction data hash
+    pub fn data_sechash(&self) -> &namada_core::hash::Hash {
+        &self.data_hash
+    }
+
+    /// Get the hash of this transaction's memo
+    pub fn memo_sechash(&self) -> &namada_core::hash::Hash {
+        &self.memo_hash
+    }
+
+    pub fn hash<'a>(&self, hasher: &'a mut Sha256) -> &'a mut Sha256 {
+        hasher.update(self.serialize_to_vec());
+        hasher
+    }
+
+    /// Get the hash of this Commitments
+    pub fn get_hash(&self) -> namada_core::hash::Hash {
+        namada_core::hash::Hash(
+            self.hash(&mut Sha256::new()).finalize_reset().into(),
+        )
+    }
+}
+
+/// A Namada transaction header indicating where transaction subcomponents can
+/// be found
+#[derive(
+    Clone,
+    Debug,
+    BorshSerialize,
+    BorshDeserialize,
+    BorshDeserializer,
+    BorshSchema,
+    Serialize,
+    Deserialize,
+    PartialEq,
+)]
+pub struct Header {
+    /// The chain which this transaction is being submitted to
+    pub chain_id: ChainId,
+    /// The time at which this transaction expires
+    pub expiration: Option<DateTimeUtc>,
+    /// A transaction timestamp
+    pub timestamp: DateTimeUtc,
+    /// The commitments to the transaction's sections
+    pub batch: HashSet<TxCommitments>,
```

Comment:
This set could be empty (no inner txs, just wrapper) and I didn't place any check on this. Should we add something?

Reply:
As in, permitting a batch of size 0? There's nothing semantically wrong with that, I think.